minerl/data/pipeline/merge.py

```diff
@@ -256,31 +269,39 @@ def main():
     if not files_to_merge:
         return
 
+    if parallel:
+        import multiprocessing
+    else:
+        import multiprocessing.dummy as multiprocessing
 
-    with multiprocessing.Pool(max(opts.num_workers, 1), tqdm.tqdm.set_lock, initargs=(multiprocessing.RLock(),)) as pool:
+    with multiprocessing.Pool(n_workers, tqdm.tqdm.set_lock, initargs=(multiprocessing.RLock(),)) as pool:
         timings = list(tqdm.tqdm(
             pool.imap_unordered(merge_stream, files_to_merge),
             total=len(files_to_merge), desc='Merging'))
         failed_streams = [x[1] for x in timings if isinstance(x, tuple)]
         times = np.array([x for x in timings if not isinstance(x, tuple)])
 
     # Write blacklist    
-    for f in (failed_streams):
+    for f in failed_streams:
         if f not in blacklist:
             blacklist.append(f)
     print(blacklist)
     with open(BLACKLIST_TXT, 'w') as f:
         f.write('\n'.join(blacklist))
 
-
     print("FINISHED WITH TIMINGS:")
     print("\t Number of streams succeeded:", len(times))
     print("\t Number of streams failed:", len(failed_streams))
     if times is not None and len(times) > 0: 
         print("\t Average time: {}".format(times.mean()))
 
 
-
+def main_console():
+    """Like main, except it also reads arguments from stdin."""
+    parser = argparse.ArgumentParser('Merge Script')
+    parser.add_argument('num_workers', type=int, help='Number of parallel workers.')
+    opts = parser.parse_args()
+    main(n_workers=opts["num_workers"])
```

Comment:
Above, you read off of sys.argv, here you use argparse. Either is fine in this case I think, but better to be consistent.

Reply:
I'll fix that up. (The discrepancy was in the original codebase; I just moved around stuff to different functions)

Reply:
I'm going to avoid converting this to parser, since they are implementing a CLI with a variable number of arguments (and it is nontrivial for me to look up how to implement cleanly in `argparse`). For our purposes, I have never used any command-line arguments while calling this script.